lib/filterx/filterx-grammar.ym

```diff
@@ -51,10 +49,11 @@
 #include "filterx/expr-condition.h"
 #include "filterx/expr-isset.h"
 #include "filterx/expr-unset.h"
+#include "filterx/expr-literal-generator.h"
 
 #include "template/templates.h"
 
-FilterXExpr *last_fillable;
+FilterXExpr *last_literal_generator;
```

Comment:
we should avoid using globals for potentially recursive grammars as when we do recurse, the 2nd iteration of the grammar will overwrite the values for the outer one.

The solution is the use of an outer token (e.g. $0), an example is visible in the template_content_inner rule in the main grammar.

I can help fixing this.

To reproduce an issue, try to embed multiple generators into each other. This may not immediately trigger an issue (depending whether we use last_literal_generator variable after recursing), but it's a disaster waiting to happen.